code/modules/mob/living/carbon/xenomorph/castes/warrior/abilities_warrior.dm

```diff
@@ -213,71 +324,166 @@
+	X.visible_message("<span class='xenodanger'>\The [X] smashes [src] with a devastating punch!</span>", \
+		"<span class='xenodanger'>We smash [src] with a devastating punch!</span>", visible_message_flags = COMBAT_MESSAGE)
+	playsound(src, pick('sound/effects/bang.ogg','sound/effects/metal_crash.ogg','sound/effects/meteorimpact.ogg'), 50, 1)
+	Shake(4, 4, 2 SECONDS)
 
-/mob/living/carbon/human/punch_act(mob/living/carbon/xenomorph/X, damage, target_zone)
-	var/datum/limb/L = get_limb(target_zone)
+	if(!CHECK_BITFIELD(machine_stat, PANEL_OPEN))
+		ENABLE_BITFIELD(machine_stat, PANEL_OPEN)
 
-	if (!L || (L.limb_status & LIMB_DESTROYED))
-		return
+	if(wires) //If it has wires, break em
+		var/allcut = wires.is_all_cut()
+		if(!allcut) //Considered prohibiting this vs airlocks, but tbh, I can see clever warriors using this to keep airlocks bolted open or closed as is most advantageous
+			wires.cut_all()
+			visible_message("<span class='danger'>\The [src]'s wires snap apart in a rain of sparks!</span>", null, null, 5)
+
+	update_icon()
+
+/obj/machinery/light/punch_act(mob/living/carbon/xenomorph/X)
+	. = ..()
+	attack_alien(X) //Smash it
+
+/obj/machinery/power/apc/punch_act(mob/living/carbon/xenomorph/X)
+	. = ..()
+	beenhit += 4 //Break it open instantly
+
+/obj/machinery/vending/punch_act(mob/living/carbon/xenomorph/X)
+	. = ..()
+	if(tipped_level < 2) //Knock it down if it isn't
+		X.visible_message("<span class='danger'>\The [X] knocks \the [src] down!</span>", \
+		"<span class='danger'>You knock \the [src] down!</span>", null, 5)
+		tip_over()
+
+/obj/structure/punch_act(mob/living/carbon/xenomorph/X, damage, target_zone) //Smash structures
+	X.do_attack_animation(src, ATTACK_EFFECT_YELLOWPUNCH)
+	X.do_attack_animation(src, ATTACK_EFFECT_DISARM2)
+	attack_alien(X, damage * 4, BRUTE, "", FALSE) //Deals 4 times regular damage to structures
+	X.visible_message("<span class='xenodanger'>\The [X] smashes [src] with a devastating punch!</span>", \
+		"<span class='xenodanger'>We smash [src] with a devastating punch!</span>", visible_message_flags = COMBAT_MESSAGE)
+	playsound(src, pick('sound/effects/bang.ogg','sound/effects/metal_crash.ogg','sound/effects/meteorimpact.ogg'), 50, 1)
+	Shake(4, 4, 2 SECONDS)
+
+/mob/living/punch_act(mob/living/carbon/xenomorph/X, damage, target_zone)
+	var/punch_description = "powerful"
+	var/stagger_stacks = 3
+	var/slowdown_stacks = 3
+
+	if(pulledby == X) //If we're being grappled by the Warrior punching us, it's gonna do extra damage and debuffs; combolicious
+		damage *= 2
+		slowdown_stacks *= 2
+		stagger_stacks *= 2
+		ParalyzeNoChain(0.5 SECONDS)
+		punch_description = "devastating"
+
+	if(iscarbon(src) && !isxeno(src)) //Xenos don't have limbs really
+		var/mob/living/carbon/carbon_victim = src
+		var/datum/limb/L = carbon_victim.get_limb(target_zone)
+
+		if (!L || (L.limb_status & LIMB_DESTROYED))
+			to_chat(X, "<span class='xenodanger'>We can't punch that which doesn't exist!</span>")
```

Comment:
I feel that the warrior in question will "in the heat of the moment" rather decide to punch some other part of the body instead of "air"

Default to the chest instead?

Reply:
sure, this is existing code but I can improve it

Reply:
Gonna have this cancel out; you probably want to reserve the cooldown and plasma for another specific target.